src/uzal_cost.jl

```diff
@@ -0,0 +1,116 @@
+using Neighborhood
+using StatsBase
+using Distances
+
+export uzal_cost
+
+
+"""
+    uzal_cost(Y; kwargs...) → L
+Compute the L-statistic `L` according to Uzal et al.[^Uzal2011],
+for a phase space trajectory `Y` (timeseries or `Dataset`). The L-statistic is
+based on theoretical arguments on noise amplification, the complexity of the
+reconstructed attractor and a direct measure of local stretch which constitutes
+an irrelevance measure.
+
+## Keyword arguments
+
+* `SampleSize = .5`: Number of considered fiducial points v as a fraction of
+  input phase space trajectory `Y`'s length, in order to average the conditional
+  variances and neighborhood sizes (read algorithm description) to produce `L`.
+* `K = 3`: the amount of nearest neighbors considered, in order to compute σ_k^2
+  (read algorithm description).
+  If given a vector, minimum result over all `k ∈ K` is returned.
+* `metric = Euclidean()`: metric used for finding nearest neigbhors in the input
+  phase space trajectory `Y.
+* `w = 1`: Theiler window (neighbors in time with index `w` close to the point,
+  that are excluded from being true neighbors). `w=0` means to exclude only the
+  point itself, and no temporal neighbors.
+* `Tw = 40`: The time horizon (in sampling units) up to which E_k^2 gets computed
+  and averaged over (read algorithm description).
+
+## Description
+tbd.
+
+[^Uzal2011]: Uzal, L. C., Grinblat, G. L., Verdes, P. F. (2011). [Optimal reconstruction of dynamical systems: A noise amplification approach. Physical Review E 84, 016223](https://doi.org/10.1103/PhysRevE.84.016223).
+"""
+function uzal_cost(Y; Tw::Int = 40, K::Int = 3, w::Int = 1, SampleSize::Float64 = .5,
```

Comment:
CamelCase naming in Julia is used for either Types or Modules, but not for variables. `SampleSize` should be made `samplesize`.

Reply:
Fair enough, will do.